worlds/tloz/Options.py

```diff
@@ -33,8 +33,68 @@ class StartingPosition(Choice):
     option_dangerous = 2
     option_very_dangerous = 3
 
+class WeaponLogic(Choice):
+    """What level of offensive power is logically required for later dungeons.
+    Easy means the Sword will be required for level 1 and later, the White Sword for levels 4 and later,
+    and the Magical Sword for levels 6, 8, and 9.
+    Moderate means the Swrod for level 1 and later and the White Sword or Magical Rod for levels 4 and later.
+    Hard means no safety logic is added. You may be required to defeat enemies with weak or unusual weaponry, such as
+    Wizzrobes with the basic Sword or Darknuts with the Magical Rod."""
+    display_name = "Weapon Logic"
+    option_easy = 0
+    option_moderate = 1
+    option_hard = 2
+
+class DefenseLogic(Choice):
+    """What level of defensive power is logically required for later dungeons.
+    Easy means you're guaranteed to have one Heart Container per previous dungeon level plus three additional,
+    and the Blue Ring is guaranteed to be available before levels 5 and later.
+    Moderate means you're guaranteed to have one extra heart per previous dungeon level,
+    taking into account available Rings.
+    Hard means no safety logic is added, so all hearts and rings could be in any location."""
+    display_name = "Defense Logic"
+    option_easy = 0
+    option_moderate = 1
+    option_hard = 2
+
+class EntranceShuffle(Choice):
+    """Shuffle entrances around.
+    Dungeons means only dungeon entrances will be shuffled with each other.
+    Major means that dungeon entrances and major item locations (sword caves, take any caves, letter cave)
+    will be shuffled with each other
+    Open means that only dungeon entrances and open caves will be shuffled with each other.
+    Major Open is a combination combines and shuffles both Major and Open locations.
+    All means all entrances will be shuffled amongst each other. Starting Sword Cave will be in an open location
+    and have a weapon.
```

Comment:
For better clarity on which ER options will force Starting Sword Cave to have a weapon:
```suggestion
    All means all entrances will be shuffled amongst each other. 
    On Open, Major Open, and All, Starting Sword Cave will be in an open location and have a weapon.
```